src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -444,27 +442,31 @@ static int lease_parse_search_domains(const uint8_t *option, size_t len, char **
                                         return r;
 
                                 n += r;
-                                continue;
                         } else if ((c & 0xc0) == 0xc0) {
                                 /* Pointer */
 
                                 uint8_t d;
                                 uint16_t ptr;
+
+                                if (pos >= len)
+                                        return -EMSGSIZE;
```

Comment:
POSIX defines EMSGSIZE as "Message too long". I don't think we should use it here. EBADMSG is more appropriate. Same comment about line 430